src/NuGet.Clients/PackageManagement.PowerShellCmdlets/Cmdlets/FindPackageCommand.cs

```diff
@@ -81,8 +81,9 @@ protected void Preprocess()
             {
                 Version = string.Empty;
             }
-
-            UpdateActiveSourceRepository(Source);
+            // Look through all available sources (including those disabled) by matching source name and url
+            var matchingSource = GetMatchingSource(Source);
+            UpdateActiveSourceRepository(Source, matchingSource);
```

Comment:
Why did you have to split the method? If validity check is optional, then consider adding a boolean flag instead to `UpdateActiveSourceRepository`.